oauthproxy.go

```diff
@@ -686,6 +717,13 @@ func (p *OAuthProxy) isAllowedRoute(req *http.Request) bool {
 	return false
 }
 
+// isForwardedRequest is used to check if X-Forwarded-Host header exists or not
+func isForwardedRequest(req *http.Request) bool {
+	gatheredHost, _ := splitHostPort(util.GetRequestHost(req))
```

Comment:
Even if we leave `X-Forwarded-Port` support, this comparison shouldn't split off the port.

If a non-standard port were present, the HTTP `Host` header would have it, right? When `GetRequestHost` doesn't have the header, it returns `req.Host`. If you splice off the port when comparing, you'd get `req.Host != gatheredHost` == `true` improperly.